### PR TITLE
User create profile

### PR DESCRIPTION
**Pull request by 123MwanjeMike**

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const mongoose = require('mongoose');
+require('../models/user');
+
+const User = mongoose.model('User');
+
+User.create = async (req, res) => {
+  try {
+    const user = new User(req.body);
+    await user.save();
+    res.status(200).send('Success');
```
Comment (nyzszn) [inline controllers/user.js:10]: Please standardize returning JSON format always,

File: controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  }
+};
+
+User.readAll = async (req, res) => {
+  try {
+    const users = await User.find();
+    res.send(users);
+  } catch (err) {
+    res.status(500).send({
+      message: err.message || 'An error occured while retrieving users',
```
Comment (nyzszn) [inline controllers/user.js:25]: Please standardize returning JSON format always

File: models/user.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const userSchema = new Schema({
+  firstname: {
+    type: String,
```
Comment (nyzszn) [inline models/user.js:7]: Please add validation on the schema level for security purposes e.g require, min, max, regex for email and phone number

File: app.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+app.use('/user', user);
+
+// Making database connection to delv
+mongoose.connect(process.env.DATABASE, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useCreateIndex: true,
+});
+
+mongoose.connection
```
Comment (nyzszn) [inline app.js:31]: Please separate concerns into individual files and methods e.g connection to database should be a file of its own.
- please use try-catch whenever possible for methods that might render an error eg. connecting to the database.

File: routes/user.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const express = require('express');
+const { User } = require('../controllers/user.js');
+
+const router = express.Router();
+
+// Create a new user
+router.post('/', User.create);
+
+// Retrieve all Customers
+router.get('/', User.readAll);
```
Comment (nyzszn) [inline routes/user.js:10]: Please add security to data-sensitive routes with Bearer tokens using **jsonwebtoken**

Comment (123MwanjeMike) [reply routes/user.js:10]: I could do this in the next sprint under user authentication, which will include login, logout, and password reset.
However, for now I've encrypted the passwords and this functionality will be next up

**Comment by 123MwanjeMike** [comment]
rename the pull request

**Pull request merged**